aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1777,7 +1777,8 @@ Tensor sparse_compressed_to_sparse(const Tensor& self, int64_t sparse_dim) {
         .mul_(at::tensor({blocksize[0], blocksize[1]}, indices.options()).reshape({2, 1}))
         .add_(at::stack(at::where(at::ones(blocksize, indices.options()))).repeat({1, nnz}));
       values = self.values().flatten(0, 2);
-      coalesced = nnz == 1;
+      // The result is always coalesced with a single block.
+      coalesced |= (plain_indices.size(-1) == 1);
```

Comment:
We always have `nnz == plain_indices.size(-1)`, so this changeset has little value.

A better estimate here is:
```suggestion
      // The result is always coalesced with a single block.
      coalesced |= (plain_indices.size(-1) == 1 || (layout == kSparseBsr && blocksize[0] == 1));
```

Reply:
Nnz in the code is COO nnz, not the block nnz.

Reply:
Ah, right.

The estimate suggestion still applies.

Reply:
Done! We can actually skip `layout == kSparseBsr` as it also applies to BSC.